idaes/core/solvers/petsc.py

```diff
@@ -360,6 +357,19 @@ def _sub_problem_scaling_suffix(m, t_block):
             if c in m.scaling_factor:
                 t_block.scaling_factor[c] = m.scaling_factor[c]
 
+class PetscDAEResults(object):
```

Comment:
Is this class really necessary? Why not just return a dictionary from ``petsc_dae_by_time_element``?

Reply:
Not really necessary, just wanted to use attributes rather than keys.